neo4j-jdbc-bolt/src/main/java/org/neo4j/jdbc/utils/BoltNeo4jUtils.java

```diff
@@ -117,4 +119,19 @@ public static void closeSafely(AutoCloseable closeable, Logger logger) {
         }
     }
 
+    public static boolean hasResultSet(BoltNeo4jConnection connection, String cypher) {
+        try (final Session session = connection.newNeo4jSession();
+             final Transaction transaction = session.beginTransaction()) {
+            return !transaction.run("EXPLAIN " + cypher)
```

Comment:
maybe avoid prefixing with `EXPLAIN ` if the (left-trimmed) query already starts with `EXPLAIN ` or `PROFILE `? feel free to add related tests for that

Reply:
not sure what to do if the query starts with `PROFILE `, though... replace `PROFILE ` with `EXPLAIN `?

Reply:
done, I just check if the query already starts with `EXPLAIN` or `PROFILE`, personally I don't like to change things under-the-hood but lemme know it you want that change